ThunkLibs/libxcb/libxcb_Guest.cpp

```diff
@@ -117,17 +129,33 @@ extern "C" {
 
   xcb_connection_t * xcb_connect(const char * a_0,int * a_1){
     auto ret = fexfn_pack_xcb_connect(a_0, a_1);
+
+    if (xcb_get_file_descriptor(ret) != -1) {
```

Comment:
XCB documentation suggests `xcb_connection_has_error` should be used for error checking.

Reply:
We could use that on connect, but we can't use that on disconnect. I opted to use the same valid display connection testing for all three instances to be consistent. I could switch the two connect ones over if you want.

Reply:
Fair enough. `xcb_get_file_descriptor` is fine then.